File: Javascript/DateOfEaster.js
const x = new Date().getFullYear();

if (Math.floor(x) != x)
{
    throw 'Year with decimal fraction is not allowed!';
}

if (x < 1583)
{
    throw 'Year earlier than 1583 is not allowed!';
}

const a = x % 19;
const b = Math.floor(x / 100)
const c = x % 100;
const d = Math.floor(b / 4);
const e = b % 4;
const f = Math.floor((b + 8) / 25);
const g = Math.floor((b - f + 1) / 3);
const h = (19 * a + b - d - g + 15) % 30;
const i = Math.floor(c / 4);
const k = c % 4;
const l = (32 + 2 * e + 2 * i - h - k) % 7;
const m = Math.floor((a + 11 * h + 22 * l) / 451);
const z = h + l - 7 * m + 114;
const n = Math.floor(z / 31);
const p = z % 31;

switch (n)
{
    case 3:
        console.log('March %d', p + 1);
        break;
    case 4:
        console.log('April %d', p + 1);
        break;
}

File: Octave/DateOfEaster.m
# Date of Easter
# Based on algorithm by Jean Meeus, Astronomical Algorithms, 1998, p. 67

while true
    x = fix(input("Enter a year:"));
    if x >= 1583
        break;
    endif
    printf("Year earlier than 1583 is not allowed!\n");
endwhile

a = rem(x, 19);
b = fix(x / 100);
c = rem(x, 100);
d = fix(b / 4);
e = rem(b, 4);
f = fix((b + 8) / 25);
g = fix((b - f + 1) / 3);
h = rem(19 * a + b - d - g + 15, 30);
i = fix(c / 4);
k = rem(c, 4);
l = rem(32 + 2 * e + 2 * i - h - k, 7);
m = fix((a + 11 * h + 22 * l) / 451);
z = h + l - 7 * m + 114;
n = int32(fix(z / 31));
p = int32(rem(z, 31)) + 1;

switch (n)
    case 3
        printf("March %d\n", p);
    case 4
        printf("April %d\n", p);
endswitch
